include/Mat.h

```diff
@@ -391,11 +483,12 @@ class Mat {
         ndim = b.ndim;
         data = b.data;
         
-        if(allocator != NULL){
+        assert(allocator==NULL);
+        /*if(allocator != NULL){ // check this again when removing assert after allocate is defined
             DimInfo dInfo(ndim, dims, strides);
             allocator->allocate(this, dInfo);
         }
```

Comment:
- Base matrix handles allocator for data block
- The non-base handle allocating their dims and strides using their own allocator, which they keep when they are pointed to a new base and need to reallocate their dims and strides.
- We should check that the destructors correctly use each deallocate with the matching allocators

Reply:
deprecated:

- base handles allocating data
- headers handle allocating meta and construct a base if needed